lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+			surface.DrawRect(0, 0, ScrW(), ScrH())
+		end
+	end)
+	hook.Add("PostDrawTranslucentRenderables", "DrawAllLasers", function()
+		for i,v in next, player.GetAll() do
+			local wep = v:GetActiveWeapon()
+			if IsValid(wep) and wep:GetClass() == "weapon_laserpointer" and wep:GetNWBool("LaserBeamOn") and wep:GetNWBool("CanDrawLaser") and v ~= LocalPlayer() then
+				local tr = v:GetEyeTrace()
+				local lsrclr = wep.LaserColors[v:GetNWFloat("LaserPower", 0)+1]
+
+				cam.Start3D(EyePos(), EyeAngles())
+					--local maxbounce = maxbounce or 0
+					local size = math.random()*10
+
+					render.SetMaterial(Material("sprites/bluelaser1"))
+					render.DrawBeam(wep:GetNWVector("laserorgpos",Vector(0,0,0,0))+wep:GetNWVector("laserorgang",Angle(0,0,0)):Up()*6, tr.HitPos, 2, 0, 12.5, lsrclr, false)
+					render.SetMaterial(Material("sprites/light_glow02_add"))
+					render.DrawQuadEasy(tr.HitPos, tr.HitNormal, size, size, lsrclr)
+
+					--[[if tr.Entity:GetMaterial() == "func_reflective_glass" and maxbounce <= 1 then--this is for the refelcted laser
+						local vec = origin-tr.HitPos
+						local normal = tr.HitNormal
+						local reflect = -2*vec:Dot(normal)*normal+vec
+						local trc = util.TraceLine({
+							start = tr.HitPos,
+							endpos = tr.HitPos + reflect:Angle():Forward() * 10000,
+							filter = false
+						})
+
+						render.SetMaterial(Material("sprites/bluelaser1"))
+						render.DrawBeam(tr.HitPos, trc.HitPos, 2, 0, 12.5, lsrclr, false)
+						render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+						render.DrawSprite(trc.HitPos, size, size, lsrclr)
+
+						maxbounce = maxbounce+1
+					end]]
+				cam.End3D()
+			end
+		end
+	end)
+	hook.Add("KeyRelease","RemoveBeam", function(ply, key)
+		if key == IN_ATTACK then
+			if IsValid(ply:GetActiveWeapon()) and ply:GetActiveWeapon():GetClass() == "weapon_laserpointer" then
+				ply:GetActiveWeapon():RemoveLaser()
+			end
+		end
+	end)
+else--server code
+	hook.Add("KeyRelease","RemoveBeam", function(ply, key)
+		if key == IN_ATTACK then
+			if IsValid(ply:GetActiveWeapon()) and ply:GetActiveWeapon():GetClass() == "weapon_laserpointer" then
+				ply:GetActiveWeapon():RemoveLaser()
+			end
+		end
+	end)
+	hook.Add("WeaponEquip", "MakeSureAmmoIs100", function(wep, ply)
+		if IsValid(wep) and wep:GetClass() == "weapon_laserpointer" then
+			if wep:Clip1() < 100 then
+				wep:SetClip1(100)
+			end
+		end
+	end)
+	hook.Add("DoPlayerDeath", "OnPlayerDeath", function(ply, att, dmg)
+        if IsValid(ply:GetActiveWeapon()) and ply:GetActiveWeapon():GetClass() == "weapon_laserpointer" then
+            ply:GetActiveWeapon():RemoveLaser()
+        end
+    end)
+    timer.Create("BlindnessUpdater", 0.1, 0, function()
+    	for i,v in next, player:GetAll() do
+    		if IsValid(v) then
+    		local blind = v:GetNWFloat("Blindness", 0)
+				if blind ~= nil then
+					if (blind or 0) > 0 then blind = blind-2 end
+					if blind < 0 then blind = 0 end
+
+					v:SetNWFloat("Blindness",blind)

```

Comment:
Don't call this unless the value has changed, i'm not sure if it networks

Reply:
SetNWFloat also will send the value to every player on the server, so it would be better to use a custom net message of some sort only send to the player in question

Reply:
okay